src/ostree/ot-builtin-commit.c

```diff
@@ -636,11 +636,22 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
             {
               if (!opt_tar_pathname_filter)
                 {
-                  object_to_commit = g_file_new_for_path (tree);
-                  if (!ostree_repo_write_archive_to_mtree (repo, object_to_commit, mtree, modifier,
-                                                           opt_tar_autocreate_parents,
-                                                           cancellable, error))
-                    goto out;
+                  if (strcmp (tree, "-") == 0)
+                    {
+                      if (!ostree_repo_write_archive_to_mtree_from_fd (repo, STDIN_FILENO, mtree, modifier,
+                                                                       opt_tar_autocreate_parents,
+                                                                       cancellable, error))
+                        goto out;
+                    }
+                  else
+                    {
+                      object_to_commit = g_file_new_for_path (tree);
```

Comment:
I briefly thought we should support a syntax like `fd=<integer>` but...eh, anyone who wants that can use `/proc/self/fd/`.

(Which on that topic, I think you can use `--tree=tar=/proc/self/fd/0` with existing libostree too, but supporting `-` is definitely a Unixy thing to do and the new API is useful)